pkg/types/core/capabilities_registry.go

```diff
@@ -52,6 +52,10 @@ func (UnimplementedCapabilitiesRegistryMetadata) ConfigForCapability(ctx context
 	return capabilities.CapabilityConfiguration{}, errors.New("ConfigForCapability not implemented")
 }
 
+func (UnimplementedCapabilitiesRegistryMetadata) TypeAndVersion(ctx context.Context) (string, string, error) {
+	return "", "", errors.New("TypeAndVersion not implemented")
```

Comment:
If you use the grpc status package, then you can include the `Unimplemented` code:
```suggestion
	return "", "", status.Errorf(codes.Unimplemented, "method TypeAndVersion not implemented")
```